### PR TITLE
refactor(tasks): removed params from prove task

### PR DESCRIPTION
**Pull request by crisgarner**
# Description

Removed some parameters from the prove tasks and added them to the deploy-config

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: contracts/tasks/runner/prove.ts [hunk 1]
```diff
@@ -66,12 +65,17 @@ task("prove", "Command to generate proof and prove the result of a poll on-chain
         // Create the directory
         fs.mkdirSync(outputDir);
       }
-
+      const config: low.LowdbSync<TConfig> = low(
+        typeof window !== "undefined"
+          ? new LocalStorageSync<TConfig>("deploy-config")
+          : new FileSync<TConfig>(path.resolve(__dirname, "..", "..", "./deploy-config.json")),
+      );
```
Comment (0xmad) [inline contracts/tasks/runner/prove.ts:72]: Lowdb should not be used outside of deployment or contract storage classes

File: contracts/tasks/runner/prove.ts [hunk 2]
```diff
@@ -99,7 +103,7 @@ task("prove", "Command to generate proof and prove the result of a poll on-chain
 
       // Check that the state and message trees have been merged for at least the first poll
       if (!isStateAqMerged && poll.toString() === "0") {
-        throw new Error("The state tree has not been merged yet. Please use the mergeSignups subcommand to do so.");
+        throw new Error("The state tree has not been merged yet. Please use the mergeSignups subcommmand to do so.");
```
Comment (0xmad) [inline contracts/tasks/runner/prove.ts:106]:
```suggestion
        throw new Error("The state tree has not been merged yet. Please use the mergeSignups subcommand to do so.");
```

File: contracts/tasks/runner/prove.ts [hunk 3]
```diff
@@ -108,7 +112,7 @@ task("prove", "Command to generate proof and prove the result of a poll on-chain
       const mainRoot = await messageAqContract.getMainRoot(messageTreeDepth.toString());
 
       if (mainRoot.toString() === "0") {
-        throw new Error("The message tree has not been merged yet. Please use the mergeMessages subcommand to do so.");
+        throw new Error("The message tree has not been merged yet. Please use the mergeMessages subcommmand to do so.");
```
Comment (0xmad) [inline contracts/tasks/runner/prove.ts:115]:
```suggestion
        throw new Error("The message tree has not been merged yet. Please use the mergeMessages subcommand to do so.");
```

File: contracts/tasks/runner/prove.ts [hunk 4]
```diff
@@ -146,20 +150,27 @@ task("prove", "Command to generate proof and prove the result of a poll on-chain
       });
       const tallyContractAddress = await tallyContract.getAddress();
 
+      let qv = "qv";
+      if (useQuadraticVoting) {
+        qv = "qv";
+      } else {
+        qv = "nonQv";
+      }
```
Comment (0xmad) [inline contracts/tasks/runner/prove.ts:158]:
```suggestion
      const mode = useQuadraticVoting ? EMode.QV : EMode.NON_QV;
```

**Review by 0xmad — CHANGES_REQUESTED** [review]
@crisgarner thanks, left some comments

File: contracts/tasks/runner/prove.ts (hunk 1 above)
Comment (0xmad) [reply contracts/tasks/runner/prove.ts:72]: And this logic is already implemented inside Deployment helper

**Comment by crisgarner** [comment]
Thanks! added the recommendations @0xmad

File: contracts/tasks/runner/prove.ts [hunk 5]
```diff
@@ -145,7 +135,16 @@ task("prove", "Command to generate proof and prove the result of a poll on-chain
...
+      const useQuadraticVoting =
+        deployment.getDeployConfigField<boolean | null>(EContracts.Poll, "useQuadraticVoting") ?? false;
+      const mode = useQuadraticVoting ? "qv" : "nonQv";
+      const tallyZkey = deployment.getDeployConfigField<string>(EContracts.VkRegistry, `zkeys.${mode}.tallyVotesZkey`);
+      const tallyWasm = deployment.getDeployConfigField<string>(EContracts.VkRegistry, `zkeys.${mode}.tallyWasm`);
+      const processZkey = deployment.getDeployConfigField<string>(
+        EContracts.VkRegistry,
+        `zkeys.${mode}.processMessagesZkey`,
+      );
+      const processWasm = deployment.getDeployConfigField<string>(EContracts.VkRegistry, `zkeys.${mode}.processWasm`);
```
Comment (0xmad) [inline contracts/tasks/runner/prove.ts:147]: For these fields you need to pass mustGet = true for `getDeployConfigField` method.

Comment (crisgarner) [reply contracts/tasks/runner/prove.ts]: fixed! thanks!

**Review by 0xmad — APPROVED** [review]
@crisgarner thanks!

**Pull request merged**